Add initial models with flyway migrations

Dpavlovski opened the pull request.

Dpavlovski committed:
added initial models

Dpavlovski commented:
Don't merge before we tested it with db.

HalidOdat commented on backend/src/main/java/org/latexscribe/LatexScribe/domain/model/Document.java:
```diff
@@ -0,0 +1,35 @@
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String filename;
+    private Long filesize;
+    @Column(name = "last_modified")
+    private LocalDateTime lastModified;
+    @Lob
+    @Column(name = "file_content")
+    private byte[] fileContent;
```
```suggestion
    @Column(name = "filename")
    private String name;
    @Column(name = "filesize")
    private Long size;
    @Column(name = "last_modified")
    private LocalDateTime lastModified;
    @Lob
    @Column(name = "file_content")
    private byte[] content;
```

I think we should remove the file prefixes from the `Document` files, this should reduce the code noise, removing the redundant name `doucment.content` instead of `doucment.fileContent`.

We can rename the fields in the database with the `@Column` to keep the naming while simplifying the usages of the fields

HalidOdat commented on backend/src/main/java/org/latexscribe/LatexScribe/domain/model/DocumentTemplate.java:
```diff
@@ -0,0 +1,28 @@
+    private Long filesize;
+    @Lob
+    @Column(name = "file_content")
+    private byte[] fileContent;
```
same here.

HalidOdat reviewed (commented):
Looks great! Just I nitpick :)

Dpavlovski committed:
added flyway migration

Dpavlovski commented:
@HalidOdat  I added the migration as well, pls check it out it works on my PC :). When i run the app it automatically create the tables in the db. Also i modified the fields you told me.

Dpavlovski commented:
Commented the security dependency since we dont need  now to log in now.

HalidOdat commented:
Works perfectly, it creates the tables correctly! So merging this :partying_face:
